liblog/tests/log.cpp

```diff
@@ -0,0 +1,205 @@
+#include "log.hpp"
+#include "doctest.h"
+#include "log_test.hpp"
+#include "utils.hpp"
+#include <boost/regex.hpp>
+#include <fstream>
+#include <future>
+#include <iostream>
+#include <sstream>
+using namespace std::string_literals;
+
+TEST_SUITE_BEGIN("liblog");
+
+TEST_CASE("liblog - log should write to a ostream correctly.") {
```

Comment:
Format of the test case name is right, but the description shouldn't be that long. It's a title not a sentence `write to ostream` would be a good name.